src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -362,23 +369,50 @@ private PaymentTransactionInfoPlugin executeInitialTransaction(final Transaction
                                                                    final Currency currency,
                                                                    final Iterable<PluginProperty> properties,
                                                                    final CallContext context) throws PaymentPluginApiException {
-        final Account account = getAccount(kbAccountId, context);
-
-        final Long transactionAmount = (amount == null ? null : amount.longValue());
-        final PaymentData paymentData = buildPaymentData(account, kbPaymentId, kbTransactionId, kbPaymentMethodId, currency, properties, context);
-        final OrderData orderData = buildOrderData(account, properties);
-        final UserData userData = buildUserData(account, properties);
-        final String termUrl = PluginProperties.findPluginPropertyValue(PROPERTY_TERM_URL, properties);
-        final SplitSettlementData splitSettlementData = null;
-        final DateTime utcNow = clock.getUTCNow();
+        killbillMetrics.markMeter(transactionType.toString());
```

Comment:
Metrics handling in `executeInitialTransaction` and `executeFollowUpTransaction` seem generic. Could we actually move it somehow to [PluginPaymentPluginApi](https://github.com/killbill/killbill-plugin-framework-java/blob/master/src/main/java/org/killbill/billing/plugin/api/payment/PluginPaymentPluginApi.java), so that all APIs are instrumented by default (rate and latency, via something like `try/finally`)?

Similarly, we could do the same for other plugin APIs (e.g. [PluginPaymentControlPluginApi](https://github.com/killbill/killbill-plugin-framework-java/blob/master/src/main/java/org/killbill/billing/plugin/api/control/PluginPaymentControlPluginApi.java)), which would give us basic metrics for free without having to instrument each plugin individually.